packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -667,6 +667,20 @@ abstract class CustomArrayNodeBase<const T extends ImplicitAllowedTypes>
 
 	protected abstract get simpleSchema(): T;
 
+	/**
+	 * Generation number which is incremented any time we have an edit on the node.
+	 * Used during iteration to make sure there has been no edits that were concurrently made.
+	 */
+	#generationNumber: number = 0;
+
+	/**
+	 * This function is only created to initialize the event listener to increment the generationNumber.
+	 * However, it should not be called anywhere else, as it will unregister generation number updates.
+	 */
+	#initializeGenerationNumberEvent = getFlexNode(this).on("nodeChanged", () => {
```

Comment:
Added constructor to `CustomArrayNodeBase` and removed `initializeGenerationNumberEvent` variable

Reply:
Craig mentioned that we should double check with you and add some comments justifying that

- This would not cause issues because of how registration goes for raw to flex nodes (I did add some test cases with unhydrated nodes)
- Never unregistering won't cause any memory leaks

Do you happen to know if there would be anything we should mention in the comments for calling this in the constructor?

`getFlexNode(this).on("nodeChanged", () => {
			this.#generationNumber += 1;
		});`

Reply:
As of a few weeks ago we will copy events from unhydrated nodes to hydrated upon insertion, so that should be okay. As for unregistering, I believe the registration will live so long as the anchornode stays alive. I don't think we have a way to dispose (in practice) flex nodes currently. If we did, you'd want to do the deregistration there. But, since we're about to do a lot of work to change the flex layer, I wouldn't worry about it here.